_posts/2022-09-21-accessibility-for-web-dev.md

```diff
@@ -0,0 +1,161 @@
+
+
+## Practically how, part 2 - manual testing
+
+Two big things (but not the only ones!) to manually test is keyboard interaction and screen reader accessibility.
+
+### Keyboard interaction
+
+First of all, it’s not only for the blind; many able-bodied people don’t use the mouse by choice as well. Having said that, I’ll repeat what many accessibility advocates say: do not hide the visual focus outline. It might be ugly, but it is useful when you’re tabbing through a page.
+
+If using pop-overs, make sure to trap the focus in it. Nothing is more annoying than having to tab-tab-tab-... through the whole page underneath to get the focus into the small centered rectangle asking you “Are you sure?”. Also keep in mind that different HTML elements have different ways of activating them, for example checkbox with space, links with enter and buttons with space again. For this very reason do not disguise links or divs to be buttons; when a user encounters a button-looking link and tries to activate it with a space, guess what happens? The page scrolls instead. How to piss off your user 101.
+
+(A quirk on Mac: you have to [turn on the keyboard navigation](https://support.apple.com/guide/mac-help/use-your-keyboard-like-a-mouse-mchlp1399/mac) in your settings.)
+
+### Screen readers
+
+Learning to use a screen reader is a bit of an adventure, but it surely makes you aware of what your app sounds like. To your surprise there might be a lot of repetition, which forces you to think of what should be labelled and how, or should some things be hidden, like purely decorative content.
+
+It’s also essential to set the application language to the correct one - otherwise you might end up with “rally English” (if your app is in English, but the screen reader uses the system language - Finnish).
+
+### Tools:
+- Your own keyboard
+- Screen readers:
+    - [Voice Over](https://webaim.org/articles/voiceover/): built-in in macOS.
+    - [Screen Reader / ChromeVox](https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn?hl=en): built-in in Chromebooks, also available as Chrome extension.
+- [Web Developer](https://chrome.google.com/webstore/detail/web-developer/bfbameneiokkgbdmiekhjnmfkcnldhhm): Chrome extension. You can disable JS, CSS, cookies, check your forms, get images’ alts visible for a quicker checkup, and a dozen other things.
+- [Accessibility Insights for Web](https://accessibilityinsights.io/docs/web/overview/): an extension for Chrome and Edge having automated checks against around 50 accessibility requirements, tabbing testing help and instructions for manual tests. Easy to create reports!
+
```

Comment:
NVDA might be the most used screen reader on Windows, maybe worth mentioning. Microsoft Narrator also.